zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/tenant/DbTenantState.java

```diff
@@ -58,17 +58,66 @@ public DbTenantState(
   public void createTenant(final TenantRecord tenantRecord) {
     tenantKey.wrapLong(tenantRecord.getTenantKey());
     tenantId.wrapString(tenantRecord.getTenantId());
-    persistedTenant.setTenant(tenantRecord);
-
+    persistedTenant
+        .setTenantKey(tenantRecord.getTenantKey())
+        .setTenantId(tenantRecord.getTenantId())
+        .setName(tenantRecord.getName());
     tenantsColumnFamily.insert(tenantKey, persistedTenant);
     tenantByIdColumnFamily.insert(tenantId, fkTenantKey);
   }
 
+  @Override
+  public void updateTenant(final TenantRecord updatedTenantRecord) {
+    tenantKey.wrapLong(updatedTenantRecord.getTenantKey());
+    final PersistedTenant persistedTenant = tenantsColumnFamily.get(tenantKey);
+
+    if (persistedTenant != null) {
+      final String oldTenantId = persistedTenant.getTenantId();
+      final String newTenantId = updatedTenantRecord.getTenantId();
+
+      if (!oldTenantId.equals(newTenantId)) {
+        tenantId.wrapString(oldTenantId);
+        tenantByIdColumnFamily.deleteExisting(tenantId);
+
+        tenantId.wrapString(newTenantId);
+        tenantByIdColumnFamily.insert(tenantId, fkTenantKey);
+      }
+
+      persistedTenant
+          .setTenantKey(updatedTenantRecord.getTenantKey())
+          .setTenantId(updatedTenantRecord.getTenantId())
+          .setName(updatedTenantRecord.getName());
+
+      tenantsColumnFamily.update(tenantKey, persistedTenant);
+    }
+  }
+
   @Override
   public Optional<TenantRecord> getTenantByKey(final long tenantKey) {
     this.tenantKey.wrapLong(tenantKey);
-    return Optional.ofNullable(tenantsColumnFamily.get(this.tenantKey))
-        .map(PersistedTenant::getTenant);
+    final PersistedTenant persistedTenant = tenantsColumnFamily.get(this.tenantKey);
+
+    if (persistedTenant != null) {
+      final TenantRecord tenantRecord = new TenantRecord();
+      tenantRecord
+          .setTenantKey(persistedTenant.getTenantKey())
+          .setTenantId(persistedTenant.getTenantId())
+          .setName(persistedTenant.getName());
+
+      // Retrieve entityKey if it exists for the tenant
+      final DbForeignKey<DbLong> fkTenantKey =
+          new DbForeignKey<>(this.tenantKey, ZbColumnFamilies.TENANTS);
```

Comment:
🤔 I don't thinlk you need to create a new `fkTenantKey`, we already wrap the `tenantKey` in a `fkTenantKey` on [line 45](https://github.com/camunda/camunda/pull/23497/files#diff-080dbb6c40fe37c9e5a7655c96d5049996025275eb1194b8947093f1fd6a8efbR45), so it should already be usable with that property (however, I would re-check this just in case 🙂).